src/lib/autoDetect.ts

```diff
@@ -0,0 +1,73 @@
+import { ExtensionContext, workspace, window, languages } from 'vscode';
+import { FileAssociations } from './types';
+const { extname } = require('path');
+
+export const initializeHubLAutoDetect = (context: ExtensionContext) => {
+  languages.setLanguageConfiguration('hubl', {
+    indentationRules: {
+      decreaseIndentPattern: /.*{%(.*?end).*%}*./,
+      increaseIndentPattern: /.*{%(?!.*end).*%}.*/,
+    },
+  });
+  const languageExtensions = {
+    '.html': 'html-hubl',
+    '.css': 'css-hubl',
+  };
+  const getHublPreference = context.workspaceState.get('USE_HUBL');
+
+  const filesConfig = workspace.getConfiguration('files');
+  const fileAssoc = filesConfig.get('associations') as FileAssociations;
+  context.workspaceState.update('USE_HUBL', null);
+
+  const showHublDetectedMessage = (extension: '.html' | '.css') => {
+    const hublDetectedMessage = (extension: '.html' | '.css') => {
+      return `It looks like this file contains HubL code. Would you like to use HubL for all ${extension} files in this project?`;
+    };
+    const hublDetectedYesButton = `Use HubL for ${extension} files`;
+    const hublDetectedNoButton = 'No';
+
+    window
+      .showInformationMessage(
+        hublDetectedMessage(extension),
+        hublDetectedYesButton,
+        hublDetectedNoButton
+      )
+      .then((selection) => {
+        if (selection === hublDetectedYesButton) {
+          updateWorkspaceFileAssociation(extension);
```

Comment:
There's a check being run on line 56-60 to see if the user has files associations set up in the workspace already. So we don't need to set the flag to TRUE because the associations being configured tells us this information. I tested also to make sure that if there were no workspace settings the extensions but they were set on the user level, that this flows through correctly as well.

Reply:
I see that - we check to see if the file association is present & return if it is - but we still continuing polling for it each time a file is opened. My suggestion is to remove the listener in that case!

Reply:
Understood! Do you think we should continue polling for it though in case the file associations are removed by the user at some point in their workspace settings?

Reply:
I would expect that if a user has manually removed the file associations themselves it'd be up to them to add it back - we have instructions on how to do it manually in the README.